edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesSpecFactory.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes
+{
+    using System.Collections.Generic;
+    using Newtonsoft.Json;
+
+    public class KubernetesSpecFactory<CombinedDockerConfig> : IKubernetesSpecFactory<CombinedDockerConfig>
+    {
+        const string DockerType = "docker";
+
+        public IList<KubernetesModule<CombinedDockerConfig>> GetSpec(IList<KubernetesModule<string>> crdSpec)
+        {
+             var newList = new List<KubernetesModule<CombinedDockerConfig>>();
+             foreach (var crdModule in crdSpec)
+             {
+                 if (crdModule.Type == DockerType)
```

Comment:
Is there a case where the type might come as DoCkEr? Would .tolower or ignorecase be useful here?

Reply:
Ignorcase makes a lot of sense.